### ISSUE TITLE
How to get back the input time series after using PatchTSTForPretraining?

### ISSUE THREAD

**Issue by nikhilajoshy** [huggingface/transformers#31195]
### System Info

-

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

My model is PatchTSTForPretraining(
  (model): PatchTSTModel(
    (scaler): PatchTSTScaler(
      (scaler): PatchTSTStdScaler()
    )
    (patchifier): PatchTSTPatchify()
    (masking): PatchTSTMasking()
    (encoder): PatchTSTEncoder(
      (embedder): PatchTSTEmbedding(
        (input_embedding): Linear(in_features=5, out_features=768, bias=True)
      )
      (positional_encoder): PatchTSTPositionalEncoding(
        (positional_dropout): Identity()
      )
      (layers): ModuleList(
        (0-11): 12 x PatchTSTEncoderLayer(
          (self_attn): PatchTSTAttention(
            (k_proj): Linear(in_features=768, out_features=768, bias=True)
            (v_proj): Linear(in_features=768, out_features=768, bias=True)
            (q_proj): Linear(in_features=768, out_features=768, bias=True)
            (out_proj): Linear(in_features=768, out_features=768, bias=True)
          )
          (dropout_path1): Identity()
          (norm_sublayer1): PatchTSTBatchNorm(
            (batchnorm): BatchNorm1d(768, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          )
          (ff): Sequential(
            (0): Linear(in_features=768, out_features=3072, bias=True)
            (1): GELUActivation()
            (2): Identity()
            (3): Linear(in_features=3072, out_features=768, bias=True)
          )
          (dropout_path3): Identity()
          (norm_sublayer3): PatchTSTBatchNorm(
            (batchnorm): BatchNorm1d(768, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          )
        )
      )
    )
  )
  (head): PatchTSTMaskPretrainHead(
    (dropout): Dropout(p=0.0, inplace=False)
    (linear): Linear(in_features=768, out_features=5, bias=True)
  )
)

prediction_output = model(time_series_data)

Output:

time_series_data = tensor([[[430.3000],
         [431.7600],
         [431.7600],
         [431.7600],
         [431.7600],
         [431.7600],
         [431.7600],
         [431.7600],
         [431.7600],
         [430.3000],
         [430.3000],
         [428.9600],
         [430.3000],
         [430.3000],
         [430.3000]]], device='cuda:0')
prediction_output = tensor([[[[-0.2321,  0.1897,  0.4731,  0.8893,  0.6723],
          [-0.5465, -0.9017,  0.0778,  0.0078,  1.3323],
          [ 0.4945,  0.5145, -0.5386, -0.7045, -1.5766],
          [ 0.2064,  0.6290, -0.8145,  1.0450, -0.2886]]]], device='cuda:0')

### Expected behavior

x_hat = self.head(model_output.last_hidden_state) produces output which is not consistent to the range of input time series values. I am trying to pretrain PatchTST for autoencoding. How do I get back the input time series?

**Comment by wishdong** [comment]
hello, have you solved the problem?
